OmiseSwift/Period.swift

```diff
@@ -0,0 +1,181 @@
+                    let ordinal = splitted.first.flatMap(MonthlyPeriodRule.Ordinal.init(ordinalString:)),
+                    let weekday = splitted.last.flatMap(Weekday.init(weekdayString:)) else {
+                        return nil
+                }
+                
+                return (ordinal, weekday)
+            })
+        
+        switch (onMonthdays, onWeekdayOfMonth) {
+        case (let onMonthdays?, nil):
+            monthlyRule = MonthlyPeriodRule.daysOfMonth(Set(onMonthdays))
+        case (nil, let onWeekdayOfMonth?):
+            monthlyRule = MonthlyPeriodRule.weekdayOfMonth(ordinal: onWeekdayOfMonth.0, weekday: onWeekdayOfMonth.1)
+        default:
+            monthlyRule = nil
+        }
+        
+        switch (period, onWeekdays, monthlyRule) {
+        case ("day", nil, nil):
+            self = .daily
+        case ("week", let onWeekdays?, nil):
+            self = .weekly(Set(onWeekdays))
+        case ("month", nil, let monthlyRule?):
+            self = .monthly(monthlyRule)
+        default:
+            return nil
+        }
+    }
+    
+    public static func ==(lhs: Period, rhs: Period) -> Bool {
+        switch (lhs, rhs) {
+        case (.daily, .daily):
+            return true
+        case (.weekly, .weekly):
+            return true
+        case (.monthly, .monthly):
+            return true
+        default:
+            return false
+        }
+    }
+}
+
+
+extension Period.MonthlyPeriodRule: Equatable {
+    /// Returns a Boolean value indicating whether two values are equal.
+    ///
+    /// Equality is the inverse of inequality. For any values `a` and `b`,
+    /// `a == b` implies that `a != b` is `false`.
+    ///
+    /// - Parameters:
+    ///   - lhs: A value to compare.
+    ///   - rhs: Another value to compare.
+    public static func ==(lhs: Period.MonthlyPeriodRule, rhs: Period.MonthlyPeriodRule) -> Bool {
+        switch (lhs, rhs) {
+        case (.daysOfMonth(let lhsDays), .daysOfMonth(let rhsDays)):
+            return lhsDays == rhsDays
+        case (.weekdayOfMonth(let lhs), .weekdayOfMonth(let rhs)):
+            return lhs.ordinal == rhs.ordinal && lhs.weekday == rhs.weekday
+        default:
+            return false
+        }
+    }
+}
+
+extension Period.Weekday {
+    init?(weekdayString: String) {
+        switch weekdayString.lowercased()  {
+        case "monday":
+            self = .monday
+        case "tuesday":
+            self = .tuesday
+        case "wednesday":
+            self = .wednesday
+        case "thursday":
+            self = .thursday
+        case "friday":
+            self = .friday
+        case "saturday":
+            self = .saturday
+        case "sunday":
+            self = .sunday
+        default:
+            return nil
+        }
+    }
+}
+
+extension Period.MonthlyPeriodRule.Ordinal {
+    init?(ordinalString: String) {
+        switch ordinalString.lowercased()  {
+        case "1st":
```

Comment:
He tole me that, `first` only be supported as a creating parameter but not gonna be return from the API. So we don't need to support it